### DISCUSSION TITLE
Adding Keyboard Shortcuts to Custom search / translation services.

### DISCUSSION THREAD

**Discussion by OrcDragonBorn** [sumatrapdfreader/sumatrapdf#3828] [General]
note: I searched multiple places for an answer to my question and couldn't find help,
also I didn't know where to ask so I searched and guessed that I'm supposed to write my question here.

I know how to add a keyboard shortcut to an existing command (for example):

Shortcuts [
	[
		Cmd = CmdOpenFolder
		Key = Alt + O
	]
]

and I know how to create a custom search service (for example):

SelectionHandlers [
  [
    URL = https://duckduckgo.com/?ia=web&q=${selection}
    Name = &DuckDuckGo
  ]
]

But I have no idea how to bind each custom search to a custom keyboard shortcut (for example in this case: binding 'Alt + d' to 'DuckDuckGo'), 
I tried adding
[
		Cmd = DuckDuckGo
		Key = Alt + d
]
to Shortcuts, and tried adding Key = Alt + d like this:
SelectionHandlers [
  [
    URL = https://duckduckgo.com/?ia=web&q=${selection}
    Name = &DuckDuckGo
    Key = Alt + d 
  ]
]

There is a line immediately after this example in the documentation in this page:
[https://www.sumatrapdfreader.org/docs/Customize-search-translation-services](url)
that says the following:
"Name is what shows in the menu. You can use & characters to add Windows hot-key for keyboard-only invocation."
Is the last sentence supposed to help me? or is it talking about something else? (so anyway I tried adding this & before and after, with and without quotation marks around the name, and many other permutations with no results, maybe I missed one?)

Is it not a feature? Am I missing something? Am I being dumb? I don't know, please help.

**Comment by GitHubRulesOK** [comment]
ok there are some issues here
You did partly correct by adding your selection handler with &Duck however see that &Deep is a default which is not always visible with right click !! you need to use `context` key then `e` then `d` then enter !! so not likely to reassign multiple keys

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/2591fe5d-3d84-4153-87d5-3a16d65a5f03)

Also the only 2 existing selection searches have a command for shortcuts
```
CmdSearchSelectionWithBing   |   | Search Selection with Bing
CmdSearchSelectionWithGoogle |   | Search Selection with Google
```

**Reply by OrcDragonBorn** [reply]
So I understand that there isn't a way to add a specific keyboard shortcut (eg. Alt + d) to a custom search service. There is a way to reach it with two keys (`Alt + e > d`), but it would be more handy if we could connect the two functions somehow (Shortcut and SelectionHandler), for example by the ability to create a new command name for the SelectionHandler then use it in the Shortcut?

**Comment by GitHubRulesOK** [comment]
@kjk 
another case where assigning more than one key and or command would help to allow single key assignments.
